### DISCUSSION TITLE
vite-plugin-ssr works when deploying on http but not on https. HELP!

### DISCUSSION THREAD

**Discussion by Dennise89** [vikejs/vike#1002] [Discussion]
I'm trying to deploy my vite-plugin-ssr app. I'm using CloudPanel on a digital ocean server.

I have a nodejs server that is running locally on the server and is exposed on port 3000. I am running the server with the command `npm run prod` and it will fire up a server that is listening to port 3000.

Before I had my certificate installed (Lets encrypt) it was working properly. Also when I remove the certificate, it works.

After installing the certificate I get this error on my site: ERR_HTTP2_PROTOCOL_ERROR.

This is my nginx config:

```
#server {
  #listen 80;
  #listen [::]:80;
  #listen 443 ssl http2;
  #listen [::]:443 ssl http2;
  #{{ssl_certificate_key}}
  #{{ssl_certificate}}
  #server_name baantje-boeken.nl;
  #return 301 https://www.baantje-boeken.nl$request_uri;
#}

server {
  listen 80;
  listen [::]:80;
  #listen 443 ssl http2;
  #listen [::]:443 ssl http2;
  #{{ssl_certificate_key}}
  #{{ssl_certificate}}
  server_name www.baantje-boeken.nl www1.baantje-boeken.nl;
  {{root}}

  {{nginx_access_log}}
  {{nginx_error_log}}

  #if ($scheme != "https") {
  #  rewrite ^ https://$host$uri permanent;
  #}

  location ~ /.well-known {
    auth_basic off;
    allow all;
  }

  {{settings}}

  index index.html;

  location / {
    proxy_pass http://127.0.0.1:{{app_port}}/;
    proxy_http_version 1.1;
    proxy_set_header X-Forwarded-Host $host;
    proxy_set_header X-Forwarded-Server $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_set_header Host $http_host;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "Upgrade";
    proxy_pass_request_headers on;
    proxy_max_temp_file_size 0;
    proxy_connect_timeout 900;
    proxy_send_timeout 900;
    proxy_read_timeout 900;
    proxy_buffer_size 128k;
    proxy_buffers 4 256k;
    proxy_busy_buffers_size 256k;
    proxy_temp_file_write_size 256k;
  }
}
```

My site is: https://www.baantje-boeken.nl. When you go to http it works.

**Comment by brillout** [comment]
I ain't that familiar with Nginx, so I can't help much here. It doesn't seem related to VPS, but let me know if you believe it does.

**Comment by mufasa71** [comment]
Had similar issue, but with AWS. Try to disable writeEarlyHints as this needs additional tweaks/support from VPS

**Reply by brillout** [reply]
> needs additional tweaks/support from VPS

Feel free to let me know if you believe VPS can improve things here.

**Reply by mufasa71** [reply]
Specially [AWS ALB not yet supports early hints](https://repost.aws/questions/QUHG1Y4PD2Rq6ycyTt13hKwQ/does-alb-target-group-support-103-early-hints)

But [CloudFlare do support it](https://blog.cloudflare.com/early-hints/)

Technology is still [experimental](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/103) though, and not all browsers support it
